flybywiresim/a32nx
# feat: support connecting mcdu to websocket server

Fixes #1616 
Fixes #6331 

## Summary of Changes

* External devices can now send/receive data from the MCDU using a websocket server.
* Added server that can be used to remotely control the MCDU from external devices and print to real printers

## Screenshots (if necessary)



![image](https://user-images.githubusercontent.com/16512508/146491472-51e03380-6482-4657-bae0-7209e8a3d89f.png)

## References





## Additional context





## Testing instructions

1. Start a flight
2. Open your community folder and navigate to `flybywire-aircraft-a320-neo\MCDU SERVER\` and open `server.exe`
3. Follow the instructions in the window that opens

Things to check:
* You are able to control the MCDU from another device on your network (the IP address given by the server is correct)
* (If you have both ethernet and wifi) You are able to connect from another device on your network when using both connections
* All MCDU pages are formatted correctly and function as intended on both the real MCDU and the external device
* Please test the MCDU client on a variety of devices and browsers


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
